packages/core/src/core.js

```
@@ -0,0 +1,225 @@
+    this.dispatch = null;
+    this.relay = null;
+  }
+
+  nodesIsEmpty() {
+    if (this.nodes == null || this.nodes.length == 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
+
+class Pocket {
+
+  constructor(opts) {
+    var blockchains = [];
+
+    if (opts.devID == null || opts.networkName == null || opts.netIDs == null || opts.version == null) {
+      throw new Error("Invalid number of arguments");
+    }
+
+    if (Array.isArray(opts.netIDs)) {
+      opts.netIDs.forEach(element => {
+        var blockchain = new Blockchain(opts.networkName, element, opts.version);
+        blockchains.push(blockchain.toJSON());
+      });
+    } else {
+      var blockchain = new Blockchain(opts.networkName, opts.netIDs, opts.version);
+      blockchains.push(blockchain.toJSON());
+    }
+
+    this.configuration = new Configuration(opts.devID, blockchains, opts.maxNodes || 5, opts.requestTimeOut || 10000);
+
+  }
+
+  createWallet(subnetwork, data) {
+    throw new Error("Must implement Create Wallet")
+  }
+  importWallet(address, privateKey, subnetwork, data) {
+    throw new Error("Must implement Import Wallet")
+  }
+
+  // Create a Relay instance
+  createRelay(blockchain, netID, version, data, devID) {
+    try {
+      return new Relay(blockchain, netID, version, data, devID, this.configuration);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  // Create a Report instance
+  createReport(ip, message) {
+    try {
+      return new Report(ip, message, this.configuration);
+    } catch (error) {
+      throw error;
+    }
+  }
+
+  // Get a Dispatch instance or create one
+  getDispatch() {
+    if (this.dispatch == null) {
+      this.dispatch = new Dispatch(this.configuration);
+    }
+    return this.dispatch;
+  }
+
+  // Filter nodes by netID and blockchain name
+  getNode(netID, network, version) {
+    try {
+      if (this.configuration.nodesIsEmpty()) {
+        throw new Error("Failed to filter nodes. List is empty.");
```

I don't feel like we should throw errors in this function, just return null if a node is not found, let the caller resolve that any way they want.